### PR TITLE
refactor(core): Use an IoC container to manage singleton classes [Part-2] (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
Continued from here https://github.com/n8n-io/n8n/pull/5509

### CODE REVIEW

**Review by flipswitchingmonkey — APPROVED** [review]

File: packages/cli/src/UserManagement/email/index.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
-import { getInstance, UserManagementMailer } from './UserManagementMailer';
+import { UserManagementMailer } from './UserManagementMailer';
```
Comment (krynble) [inline packages/cli/src/UserManagement/email/index.ts:1]: Maybe we can remove this barrel file - it's here but adds no value WDYT?

**Review by krynble — APPROVED** [review]
LGTM, left a few comments with questions and ideas.

**Comment by janober** [comment]
Got released with [`n8n@0.221.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.221.2)

**Comment by janober** [comment]
Got released with [`n8n@0.221.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.221.2)

**Pull request merged**